scripts/monthly_means.py

```diff
@@ -69,37 +80,47 @@ def process_month(files: list[Path], product: str) -> xr.Dataset:
 
     # Initialize all variables to zero and create a count variable
     variables = set(ds.variables) - set(ds.coords)
+
+    float_type = np.float32
+    if precision == "double":
+        float_type = np.float64
+
     for v in variables:
         # float instead of float32 to avoid any limitations accumulating
         # .copy to assign coordinates correctly
         ds[v] = ds[v].copy(
             data=np.zeros_like(ds[v]), deep=True
-        ).astype(float)
+        ).astype(float_type)
         ds[f'{v}_count'] = ds[v].copy(
             data=np.zeros_like(ds[v]), deep=True
-        ).astype(int)
+        ).astype(np.int16)
 
     # Accumulate values
-    for f in tqdm(files, dynamic_ncols=True):
-        ds_f = xr.open_zarr(f).load()
+    itr = files
+    if status_bar:
+        itr = tqdm(files, dynamic_ncols=True)
+
+    for path in itr:
+        ds_f = xr.load_dataset(path)
 
         with warnings.catch_warnings():
             warnings.filterwarnings("ignore", message=(
                 "Converting non-nanosecond precision"
-                )
             )
+                                    )
```

Comment:
With the current format, I spent a few minutes trying to understand this 'change'. Did `black` format this bit?
Anyhow, OK.